src/amrex/StructOfArrays.py

```diff
@@ -0,0 +1,99 @@
+"""
+This file is part of pyAMReX
+
+Copyright 2023 AMReX community
+Authors: Axel Huebl
+License: BSD-3-Clause-LBNL
+"""
+from collections import namedtuple
+
+
+def soa_to_numpy(self, copy=False):
+    """
+    Provide Numpy views into a StructOfArrays.
+
+    Parameters
+    ----------
+    self : amrex.StructOfArrays_*
+        A StructOfArrays class in pyAMReX
+    copy : bool, optional
+        Copy the data if true, otherwise create a view (default).
+
+    Returns
+    -------
+    namedtuple
+        A tuple with real and int components that are each lists
+        of 1D numpy arrays.
+    """
+    import numpy as np
+
+    SoA_np = namedtuple(type(self).__name__ + "_np", ["real", "int"])
+
+    soa_view = SoA_np([], [])
+
+    if self.size() == 0:
+        raise ValueError("SoA is empty.")
+
+    for idx_real in range(self.NumRealComps()):
+        soa_view.real.append(self.GetRealData(idx_real).to_numpy(copy=copy))
+
+    for idx_int in range(self.NumIntComps()):
+        soa_view.int.append(self.GetIntData(idx_int).to_numpy(copy=copy))
```

Comment:
Double check if runtime attributes are part of this:
https://github.com/AMReX-Codes/pyamrex/pull/220#discussion_r1382548333